truffle/src/main/java/org/jruby/truffle/options/StringArrayOptionDescription.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright (c) 2016 Oracle and/or its affiliates. All rights reserved. This
+ * code is released under a tri EPL/GPL/LGPL license. You can use it,
+ * redistribute it and/or modify it under the terms of the:
+ *
+ * Eclipse Public License version 1.0
+ * GNU General Public License version 2
+ * GNU Lesser General Public License version 2.1
+ */
+package org.jruby.truffle.options;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class StringArrayOptionDescription extends OptionDescription {
+
+    private final String[] defaultValue;
+
+    public StringArrayOptionDescription(String name, String description, String[] defaultValue) {
+        super(name, description);
+        this.defaultValue = defaultValue;
+    }
+
+    @Override
+    public Object getDefaultValue() {
+        return defaultValue.clone();
+    }
+
+    @Override
+    public Object checkValue(Object value) {
+        if (value instanceof String[]) {
+            return value;
+        } else if (value instanceof String) {
+            return parseStringArray((String) value);
+        } else {
+            throw new OptionTypeException();
+        }
+    }
+
+    // Allows input such as [foo, "bar", 'baz']. Doesn't support escape sequences.
+
+    private String[] parseStringArray(String string) {
```

Comment:
I guess that wouldn't help the argv case, so maybe a value parser is still necessary. It just seems a bit complex.